app.py

```diff
@@ -51,35 +78,84 @@ def _existing_producer(queue: Queue, build_watcher_namespace: str):
         _LOGGER.debug("Listing tags available for %r", item["metadata"]["name"])
         for tag_info in item.status.tags or []:
             output_reference = f"{item.status.dockerImageRepository}:{tag_info.tag}"
+            _METRIC_IMAGES_SUBMITTED.inc()
             _LOGGER.info("Queueing already existing image %r for analysis", output_reference)
+            try:
+                _LOGGER.info(f"Submitting metrics to Prometheus pushgateway {_THOTH_METRICS_PUSHGATEWAY_URL}")
+                push_to_gateway(_THOTH_METRICS_PUSHGATEWAY_URL, job="build-watcher", registry=prometheus_registry)
+            except Exception as e:
+                _LOGGER.exception(f"An error occurred pushing the metrics: {str(e)}")
             queue.put(output_reference)
 
     _LOGGER.info("Queuing existing images for analyses has finished, all of them were scheduled for analysis")
 
 
+def _get_build(openshift, strategy: dict, build_reference: dict, event_metadata: dict) -> dict:
+    """Gather Build log and Base Image based upon the strategy of the build."""
+    if strategy.get("sourceStrategy"):
+        build_reference["base_input_reference"] = strategy.get("sourceStrategy", {}).get("from", {}).get("name", "")
+    elif strategy.get("dockerStrategy") and strategy.get("dockerStrategy", {}).get("from", {}):
+        build_reference["base_input_reference"] = strategy.get("dockerStrategy", {}).get("from", {}).get("name", "")
+    else:
+        build_reference["base_input_reference"] = ""
+    try:
+        build_log = openshift.get_build_log(
```

Comment:
Use `None` to signalize there is no value (same for `build_log`) so it can be distinguished if the value is an empty string or really nothing (none). The openapi schema on user-api should state nullable values for that case.